packages/profile-sync-controller/src/AuthenticationController.ts

```diff
@@ -0,0 +1,338 @@
+    handler: AuthenticationController[K];
+  };
+};
+type ActionsObj = CreateActionsObj<
+  | 'performSignIn'
+  | 'performSignOut'
+  | 'getBearerToken'
+  | 'getSessionProfile'
+  | 'isSignedIn'
+>;
+export type Actions = ActionsObj[keyof ActionsObj];
+export type AuthenticationControllerPerformSignIn = ActionsObj['performSignIn'];
+export type AuthenticationControllerPerformSignOut =
+  ActionsObj['performSignOut'];
+export type AuthenticationControllerGetBearerToken =
+  ActionsObj['getBearerToken'];
+export type AuthenticationControllerGetSessionProfile =
+  ActionsObj['getSessionProfile'];
+export type AuthenticationControllerIsSignedIn = ActionsObj['isSignedIn'];
+
+// Allowed Actions
+export type AllowedActions =
+  | HandleSnapRequest
+  | UserStorageControllerDisableProfileSyncing;
+
+// Messenger
+export type AuthenticationControllerMessenger = RestrictedControllerMessenger<
+  typeof controllerName,
+  Actions | AllowedActions,
+  never,
+  AllowedActions['type'],
+  never
+>;
+
+/**
+ * Controller that enables authentication for restricted endpoints.
+ * Used for Global Profile Syncing and Notifications
+ */
+export class AuthenticationController extends BaseController<
+  typeof controllerName,
+  AuthenticationControllerState,
+  AuthenticationControllerMessenger
+> {
+  #metametrics: MetaMetricsAuth;
+
+  constructor({
+    messenger,
+    state,
+    metametrics,
+  }: {
+    messenger: AuthenticationControllerMessenger;
+    state?: AuthenticationControllerState;
+    /**
+     * Not using the Messaging System as we
+     * do not want to tie this strictly to extension
+     */
+    metametrics: MetaMetricsAuth;
+  }) {
+    super({
+      messenger,
+      metadata,
+      name: controllerName,
+      state: { ...defaultState, ...state },
+    });
+
+    this.#metametrics = metametrics;
+
+    this.#registerMessageHandlers();
+  }
+
+  listAccounts(): InternalAccount[] | undefined {
+    return (
+      this.state.internalAccounts &&
+      Object.values(this.state.internalAccounts.accounts)
+    );
+  }
+
+  /**
+   * Constructor helper for registering this controller's messaging system
+   * actions.
+   */
+  #registerMessageHandlers(): void {
+    this.messagingSystem.registerActionHandler(
+      'AuthenticationController:getBearerToken',
+      this.getBearerToken.bind(this),
+    );
+
+    this.messagingSystem.registerActionHandler(
+      'AuthenticationController:getSessionProfile',
+      this.getSessionProfile.bind(this),
+    );
+
+    this.messagingSystem.registerActionHandler(
+      'AuthenticationController:isSignedIn',
+      this.isSignedIn.bind(this),
+    );
+
+    this.messagingSystem.registerActionHandler(
+      'AuthenticationController:performSignIn',
+      this.performSignIn.bind(this),
+    );
+
+    this.messagingSystem.registerActionHandler(
+      'AuthenticationController:performSignOut',
+      this.performSignOut.bind(this),
+    );
+  }
+
+  public async performSignIn(): Promise<string> {
```

Comment:
Similar comment as for the other controller — anything that doesn't have a `#` in front of it is public. Thoughts on removing the `public` keyword?

Reply:
addressed